src/UI/Components/JSONEditor/JSONEditor.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Editor, OnValidate, useMonaco } from "@monaco-editor/react";
+import { Alert, Panel, Spinner } from "@patternfly/react-core";
+import { InfoAltIcon } from "@patternfly/react-icons";
+import styled from "styled-components";
+import { useGetJSONSchema } from "@/Data/Managers/V2/GetJSONSchema";
+import { DependencyContext } from "@/UI";
+
+interface Props {
+  service_entity: string;
+  data: string;
+  onChange: (value: string) => void;
+}
+
+/**
+ * JSON Editor component.
+ *
+ * @props {Props} props - The props of the component.
+ *  @prop {string} service_entity - The service entity.
+ *  @prop {string} data - The data to be displayed in the editor.
+ *  @prop {function} onChange - Callback method when the data is changed in the editor.
+ *
+ * @notes The JSON Editor component uses the Monaco Editor, React version.
+ * for more information, see https://www.npmjs.com/package/@monaco-editor/react?activeTab=readme
+ * for the monaco editor API, see https://microsoft.github.io/monaco-editor/typedoc/index.html
+ *
+ * @returns {React.ReactElement} The JSON Editor component.
+ */
+export const JSONEditor: React.FC<Props> = ({
+  service_entity,
+  data,
+  onChange,
+}) => {
+  const { environmentHandler } = useContext(DependencyContext);
+  const environment = environmentHandler.useId();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [editorState, setEditorState] = useState<string>(data);
+  const [errors, setErrors] = useState<string[]>([]);
+
+  const schema = useGetJSONSchema(service_entity, environment).useOneTime();
+
+  const monaco = useMonaco();
+
+  // local method to update state when the editor content changes
+  const handleEditorChange = (value: string | undefined, _event) => {
+    value && setEditorState(value);
+  };
+
+  /**
+   *
+   * @param markers - The markers from the monaco editor.
+   * It will set the errors state with the error messages.
+   * see : https://microsoft.github.io/monaco-editor/typedoc/interfaces/editor.IMarker.html
+   *
+   * @return {void}
+   */
+  const handleOnValidate: OnValidate = (markers) => {
+    if (markers.length === 0) {
+      setErrors([]);
+    }
```

Comment:
nothing big but we could do else there as if length is 0 then there is no markers to iterate through

Reply:
done